core/templates/components/smooth-height/smooth-height.component.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the smooth height change animation.
```

Comment:
It's a bit hard to tell what this does from the name of the file. Perhaps smooth-height-animation.component.ts -- is there any standard in Angular for how to name "animation" components? Should we also be putting it in an animations folder? /cc @srijanreddy98

Reply:
Changed to `smooth-height-animation.component.ts`.
There is no concept of "animation" components in angular, this is a workaround that works when we want to animate to a dynamic height.